### ISSUE TITLE
[QUESTION] APP Notification

### ISSUE THREAD

**Issue by chyok** [RocketChat/Rocket.Chat.Android#690]
could you please tell me why I have no globe message notification when run in background

**Comment by rafaelks** [comment]
@ChenHaoyuan1995 What you mean by `globe message notification`?

**Comment by chyok** [comment]
@rafaelks Thanks for reply.
      When running in the backstage, it seems that new messages received by the app could not be notified to the user. Only if you open the app can you see the messages. How could I make the messages informed to the user just like short messages?

**Comment by rafaelks** [comment]
@ChenHaoyuan1995 I'm not sure I understood... view the message how? What's a `globe message notification`?

**Comment by chyok** [comment]
@rafaelks I'm sorry I'm not expressing exactly enough, I seem to have created a non-existent term "globe message notification"..Actually what I want to know is how can I receive a message in the background phone will remind me...Thanks for reply again :)

**Comment by rafaelks** [comment]
@ChenHaoyuan1995 We don't leave the socket opened if the app is in background. What happens is that your app will receive a notification. You can press this notification to go to the message.

Does that answer your question?

**Comment by chyok** [comment]
@rafaelks How could I open  the socket when app is in background..I prefer a reminder function when app 
receive a message in background ..

**Comment by mineZcash** [comment]
I believe that this user is referring to push notifications being broken. I'm still having this issue. I'm unsure if it is the Cordova app or the Rocket.chat but no users are receiving push notifications for @ mentions (except for the Native app) nor are new DM channels pushed. Users get the email for a new DM before they get any other notification.

Logs show: 
`[34mI20180110-14:04:22.966(0) removing push token eUwwR_gntvU:APA91bHR78CuT49BjSIqtkd8VbvYQkPGYdUGB3BP8KU3xL0TFbHvF7au46aCUMbs07me_6H8kESf9ueBbJn6FPNVATzLHQ_5AWxD_-T6mtowT7WjMYqfEdSsnmbZslkWe0AUIQQck6pm 
[34mI20180110-14:04:22.967(0) removing push token cDOUNgWe_GE:APA91bHZGQFHm5AB4hr3tFcTYog1cQZRsLTihIdS04XiAv8hGeiE4q-HYzOBgsD9GVL1t3o4kAWB8Lc9hsre_K4AAkQt0EHlm1C8DvqsnW-infK-BV7ILDuOM-ulJOorjp7AV-uPJnx7 
[34mI20180110-20:32:31.145(0) removing push token e_VW_WMWk7s:APA91bH2b8aUYAvHf8qzw861D6WH_XvPXgWwgdPGxLyynqjQ2lxcTMpGtg4dKhKezMBdEAcw1P66NLYQA88STyrRK01P41zc6bEp7IH6tgNmpochDxvb3Shc3jGBmI7Su7lppK49884T`

Is there a reason why push notification tokens are being automatically removed? Is there a setting that I'm missing?

My chat : https://www.chat.zcashcommunity.com

**Comment by rafaelks** [comment]
@mineZcash Thank you for your feedback. Regarding the tokens being removed: this happens when the token fails to send a push token because it is invalid.

Quick questions: to send push notifications, are you guys using our gateway? Does the push notification works to any app (iOS, Android, Cordova)?

**Comment by rafaelks** [comment]
@ChenHaoyuan1995 You can't right now. We stop the service to save your battery's life.

**Comment by mineZcash** [comment]
Yes, notifications work on the iOS native app and the Android Native app. Notifications don't work on Cordova (iOS or Android) or in mobile Browser. I'm sending you a PM in the chat @rafaelks

**Comment by mineZcash** [comment]
@rafaelks I think I've narrowed this issue down to the Cordova app not refreshing cache properly when the app has been idle.  

I have both, Cordova and Native app installed on Android 7.0. I can check the Cordova app and the Native app to see the same unread channels and DMs in both apps. Then if I don't check for an hour or so I can re-open the Native app and upon connection it will show the unread channels, but the Cordova app will not. 

Just a few minutes ago I got a DM, which came as a push notification for both apps, from a new user I had never seen before, then upon checking the notification from the Cordova app the users DM is not visible, it even says "user doesn't exist" if I click on thier username in the main channel. I can't even see the new private channel for the user. Then checking the Native app, I can see the new users DM fine. 

To fix this in the Cordova app I have to go into app settings and clear the cache and app data, then after re-sign in and re-connect it will show the users DM. 

Should I open this as a separate bug?
